### ISSUE TITLE
Tripal 4 error on "Tripal Content Terms" page

### ISSUE THREAD

**Issue by dsenalik** [tripal/tripal#1518]
# BUG/ERROR report

### Issue description and steps to reproduce
Go to the "Tripal Content Terms" page at
`/admin/tripal/config/terms`
and see this error

The website encountered an unexpected error. Please try again later.

_Symfony\Component\Routing\Exception\MissingMandatoryParametersException_: Some mandatory parameters are missing ("config") to generate a URL for route "entity.tripal_content_terms.devel_load". in _Drupal\Core\Routing\UrlGenerator->doGenerate()_ (line _181_ of _core/lib/Drupal/Core/Routing/UrlGenerator.php_). 
```
Drupal\Core\Routing\UrlGenerator->getInternalPathFromRoute('entity.tripal_content_terms.devel_load', Object, Array, Array) (Line: 291)
Drupal\Core\Routing\UrlGenerator->generateFromRoute('entity.tripal_content_terms.devel_load', Array, Array, 1) (Line: 105)
Drupal\Core\Render\MetadataBubblingUrlGenerator->generateFromRoute('entity.tripal_content_terms.devel_load', Array, Array, 1) (Line: 765)
Drupal\Core\Url->toString(1) (Line: 182)
Drupal\Core\Utility\LinkGenerator->generate(Object, Object) (Line: 95)
Drupal\Core\Render\Element\Link::preRenderLink(Array)
call_user_func_array(Array, Array) (Line: 101)
Drupal\Core\Render\Renderer->doTrustedCallback(Array, Array, 'Render #pre_render callbacks must be methods of a class that implements \Drupal\Core\Security\TrustedCallbackInterface or be an anonymous function. The callback was %s. See https://www.drupal.org/node/2966725', 'exception', 'Drupal\Core\Render\Element\RenderCallbackInterface') (Line: 788)
Drupal\Core\Render\Renderer->doCallback('#pre_render', Array, Array) (Line: 374)
Drupal\Core\Render\Renderer->doRender(Array, ) (Line: 204)
Drupal\Core\Render\Renderer->render(Array) (Line: 477)
Drupal\Core\Template\TwigExtension->escapeFilter(Object, Array, 'html', NULL, 1) (Line: 77)
__TwigTemplate_34946472be35ccd4518fe76ed610006d->doDisplay(Array, Array) (Line: 394)
Twig\Template->displayWithErrorHandling(Array, Array) (Line: 367)
Twig\Template->display(Array) (Line: 379)
Twig\Template->render(Array, Array) (Line: 40)
Twig\TemplateWrapper->render(Array) (Line: 53)
twig_render_template('core/modules/system/templates/links.html.twig', Array) (Line: 372)
Drupal\Core\Theme\ThemeManager->render('links__dropbutton', Array) (Line: 433)
Drupal\Core\Render\Renderer->doRender(Array, ) (Line: 204)
Drupal\Core\Render\Renderer->render(Array) (Line: 477)
Drupal\Core\Template\TwigExtension->escapeFilter(Object, Array, 'html', NULL, 1) (Line: 179)
__TwigTemplate_b7b6763693d63c9e8dbc6b54b96b0a08->doDisplay(Array, Array) (Line: 394)
Twig\Template->displayWithErrorHandling(Array, Array) (Line: 367)
Twig\Template->display(Array) (Line: 379)
Twig\Template->render(Array, Array) (Line: 40)
Twig\TemplateWrapper->render(Array) (Line: 53)
twig_render_template('core/themes/claro/templates/classy/dataset/table.html.twig', Array) (Line: 372)
Drupal\Core\Theme\ThemeManager->render('table', Array) (Line: 433)
Drupal\Core\Render\Renderer->doRender(Array) (Line: 446)
Drupal\Core\Render\Renderer->doRender(Array, ) (Line: 204)
Drupal\Core\Render\Renderer->render(Array, ) (Line: 238)
Drupal\Core\Render\MainContent\HtmlRenderer->Drupal\Core\Render\MainContent\{closure}() (Line: 580)
Drupal\Core\Render\Renderer->executeInRenderContext(Object, Object) (Line: 239)
Drupal\Core\Render\MainContent\HtmlRenderer->prepare(Array, Object, Object) (Line: 128)
Drupal\Core\Render\MainContent\HtmlRenderer->renderResponse(Array, Object, Object) (Line: 90)
Drupal\Core\EventSubscriber\MainContentViewSubscriber->onViewRenderArray(Object, 'kernel.view', Object)
call_user_func(Array, Object, 'kernel.view', Object) (Line: 111)
Drupal\Component\EventDispatcher\ContainerAwareEventDispatcher->dispatch(Object, 'kernel.view') (Line: 168)
Symfony\Component\HttpKernel\HttpKernel->handleRaw(Object, 1) (Line: 74)
Symfony\Component\HttpKernel\HttpKernel->handle(Object, 1, 1) (Line: 58)
Drupal\Core\StackMiddleware\Session->handle(Object, 1, 1) (Line: 48)
Drupal\Core\StackMiddleware\KernelPreHandle->handle(Object, 1, 1) (Line: 106)
Drupal\page_cache\StackMiddleware\PageCache->pass(Object, 1, 1) (Line: 85)
Drupal\page_cache\StackMiddleware\PageCache->handle(Object, 1, 1) (Line: 48)
Drupal\Core\StackMiddleware\ReverseProxyMiddleware->handle(Object, 1, 1) (Line: 51)
Drupal\Core\StackMiddleware\NegotiationMiddleware->handle(Object, 1, 1) (Line: 51)
Drupal\Core\StackMiddleware\StackedHttpKernel->handle(Object, 1, 1) (Line: 686)
Drupal\Core\DrupalKernel->handle(Object) (Line: 19)
```

**Comment by Ferrisx4** [comment]
Did a little testing on this, it seems to be caused by interference from the Devel module, if it is enabled. Disabling devel causes the page to load correctly, while still giving a similar error in the log.

Another module experienced a similar issue, but their simple fix was not enough in this situation: https://www.drupal.org/project/theme_switcher/issues/3307042#comment-15176445

**Comment by dsenalik** [comment]
No, I think you have found the solution. Like your linked patch, we need to do the same, replacing `{config}` with the name from the `id` field. Documentation on this:
https://www.drupal.org/docs/drupal-apis/configuration-api/creating-a-configuration-entity-type
There are a couple of other instances that, although they seem to work, should be fixed too.
This is all in pull #1597

**Comment by Ferrisx4** [comment]
I'm glad it worked for you @dsenalik! @spficklin and I had tested this and could not get it to work. I will test your patch in the morning, but it looks good.

**Comment by dsenalik** [comment]
Fixed with merged PR #1597
